pkg/tests/upgrade_tests/upgrade_utils.go

```diff
@@ -0,0 +1,221 @@
+			t.Logf("functions\nexpected:\n\t%s\ngot:\n\t%s", expected.functions, schema.functions)
+		}
+		if privilegesDiff {
+			t.Logf("privileges\nexpected:\n\t%s\ngot:\n\t%s", expected.privileges, schema.privileges)
+		}
+		if indicesDiff {
+			t.Logf("indices\nexpected:\n\t%s\ngot:\n\t%s", expected.indices, schema.indices)
+		}
+		if triggersDiff {
+			t.Logf("triggers\nexpected:\n\t%s\ngot:\n\t%s", expected.triggers, schema.triggers)
+		}
+		if dataDiff {
+			t.Logf("data\nexpected:\n\t%+v\ngot:\n\t%+v", expected.data, schema.data)
+		}
+	}
+}
+
+var replaceChildren = regexp.MustCompile("timescaledb_internal\\._hyper_.*\n")
+
+func GetDbInfo(t *testing.T, container testcontainers.Container, dbName, outputDir string, db *pgxpool.Pool) (info dbInfo) {
+
+	info.schemaNames = getSchemas(t, db)
+	if !reflect.DeepEqual(info.schemaNames, schemas) {
+		t.Errorf(
+			"unexpected schemas.\nexpected\n\t%v\ngot\n\t%v",
+			info.schemas,
+			schemas,
+		)
+	}
+
+	info.extensions = getPsqlInfo(t, container, dbName, outputDir, "\\dx")
+	info.schemas = make([]schemaInfo, len(ourSchemas))
+	for i, schema := range ourSchemas {
+		info := &info.schemas[i]
+		info.name = schema
+		info.tables = getPsqlInfo(t, container, dbName, outputDir, "\\d+ "+schema+".*")
+		info.tables = replaceChildren.ReplaceAllLiteralString(info.tables, "timescaledb_internal._hyper_*\n")
+		info.functions = getPsqlInfo(t, container, dbName, outputDir, "\\df+ "+schema+".*")
+		info.privileges = getPsqlInfo(t, container, dbName, outputDir, "\\dp "+schema+".*")
+		// not using \di+ since the sizes are too noisy, and the descriptions
+		// will be in tables anyway
+		info.indices = getPsqlInfo(t, container, dbName, outputDir, "\\di "+schema+".*")
+		info.triggers = getPsqlInfo(t, container, dbName, outputDir, "\\dy "+schema+".*")
+		info.data = getTableInfosForSchema(t, db, schema)
+	}
+	return
+}
+
+func getSchemas(t *testing.T, db *pgxpool.Pool) (out []string) {
+	row := db.QueryRow(
+		context.Background(),
+		"SELECT array_agg(nspname::TEXT order by nspname::TEXT) FROM pg_namespace",
+	)
+	err := row.Scan(&out)
+	if err != nil {
+		t.Errorf("could not discover schemas due to: %v", err)
+	}
+	return
+}
+
+func getPsqlInfo(t *testing.T, container testcontainers.Container, dbName string, outputDir string, query string) string {
+	i, err := container.Exec(
+		context.Background(),
+		[]string{"bash", "-c", "psql -U postgres -d " + dbName + " -c '" + query + "' &> /testdata/output.out"},
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	output := readOutput(t, outputDir)
+
+	if i != 0 {
+		t.Logf("psql error. output: %s", output)
+	}
+	return output
+}
+
+func readOutput(t *testing.T, outputDir string) string {
+	outputFile := outputDir + "/output.out"
+	output, err := ioutil.ReadFile(outputFile)
+	if err != nil {
+		t.Errorf("error reading psql output: %v", err)
+	}
+	return string(output)
+}
+
+func getTableInfosForSchema(t *testing.T, db *pgxpool.Pool, schema string) (out []tableInfo) {
```

Comment:
For some reason, the before and after state miss-match at here. Any suggestions would be very useful.

Reply:
ok, so there are some changes in label_key table which are fine and should just be ignored. 

there are some changes in label_key_position table. That should be fixed. We should delete all rows for the metric_name being deleted in those tables.